### DISCUSSION TITLE
Hinged Base Connection Template cannot be applied through API

### DISCUSSION THREAD

**Discussion by Ahmad-Ghuneim** [idea-statica/ideastatica-public#636] [Help Q&A]
Hello,

I am trying to apply a Hinged Base Template to a project connection through API, I successfully created a file through IOM xml & xmlR files, but when I apply the Hinged Base Template the resulting string is "Error", however, other simpler templates can be applied through API easily.

Hinged base template can be applied with no issues internally from Propose tab.

result from hinged base template 
![image](https://github.com/user-attachments/assets/d1191121-ed6d-4010-ad6a-1344d0d84664)

result from simple web opening template 
![image](https://github.com/user-attachments/assets/f5820549-6a0c-4d90-9a1a-d856b22b65a7)

Connection created from IOM files 
![image](https://github.com/user-attachments/assets/abcb0f2a-379c-45fb-bc68-bbaa6f54def1)

Intended Hinged Base template to be applied
![image](https://github.com/user-attachments/assets/4e460ffb-a91a-48e3-b367-137789d1f25d)

V24.0.4

Regards

**Comment by nathanlukeIDEA** [comment]
Hi @Ahmad-Ghuneim,

I suspect it is because of the mapping of the concrete material for the concrete block. You likely do not have a concrete material in the project file generated from the IOM.

Can you please try changing the apply con template settings. If you have a standard idea concrete material set in the template and you set `applysettings.UseMatFromOrigin = true `it should attempt to import the concrete material.

If not please send me your IOM file and template file and I will test it.

Cheers
Nathan

**Comment by Ahmad-Ghuneim** [comment]
Hi @nathanlukeIDEA,

The IOM represents only a vertical member with no operations, and I can't find any concrete material in the generated IOM,

In the template file, I use the default 4000psi grade for concrete block, and default 5/8 F1554 Gr.36 for anchor bolt, and `applysettings.UseMatFromOrigin = true` still gives me an error.

attached are the IOM model and results files along with the .contemp template file.
[Hinged Base Application.zip](https://github.com/user-attachments/files/16788608/Hinged.Base.Application.zip)

Cheers,
Ghuneim

**Comment by Ahmad-Ghuneim** [comment]
Hi @nathanlukeIDEA 

The main issue was that the Member name was not the same in the IOM file and the Template file, after renaming the name in the IOM file, it worked, even though it's not mandatory when applying the template from inside ideaStatica application.

Cheers,
Ghuneim
